arrow/util.py

```diff
@@ -46,6 +52,15 @@ def is_timestamp(value):
         return False
 
 
+def is_ordinal(value):
+    """Check if value is a valid ordinal."""
+    if isinstance(value, bool) or not isinstance(value, int):
+        raise TypeError(f"ordinal must be an intger (got type {type(value)})")
+    if not (MIN_ORDINAL <= value <= MAX_ORDINAL):
+        raise ValueError(f"ordinal {value} is out of range.")
+    return value
```

Comment:
This should probably be refactored to a function that returns a boolean to better correspond to the function name. If the value is not a valid type or not within range, return false.

Reply:
Thanks for the feedback!

Yes that is how I had planned it initially, but then I noticed that datetime raises ValueError for out of range inputs and TypeError for incorrect types, so I thought maybe we should raise the specific exceptions too, from this function itself, because once I return I won't know which exception to raise? If we keep it as is then I should probably rename the function and edit the function descriptor as well.

Another option could be to return different error codes for each case (TypeError versus ValueError) instead of just returning False. And then based on the error code I can raise the exception in from_ordinal.

And then there is also the option to just raise ValueError for any invalid input.

Is there a preference?

Reply:
I see. Maybe call the function `validate_ordinal(value)`, and instead of returning the value, it is just used to validate the ordinal and raise the appropriate error. What do you think? I think making this a void function dedicated to validation makes more sense.

Reply:
Yes I think this will be a good option and validate_ordinal is a more appropriate name. I changed the implementation to reflect our discussion and it is pushed now. You can let me know if you have any additional comments. Thanks for reviewing this PR!